### PR TITLE
Allow unprotect resources from protectedResourceMap on certain http methods

### PR DESCRIPTION
**Pull request by at-DanielVargas**
Allow unprotect resources from protectedResourceMap on certain http methods

### CODE REVIEW

File: lib/msal-angular/angular.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
+  "cli": {
+    "analytics": "680523ab-b3dc-475f-84ea-403f7164c131"
+  },
```
Comment (jasonnutter) [inline lib/msal-angular/angular.json:5]: Please remove.

File: lib/msal-angular/src/msal.interceptor.spec.ts [hunk 2]
```diff
@@ -320,6 +321,16 @@ describe('MsalInterceptor', () => {
     done();
   });
 
+  it("does not attach authorization header when scopes set to null on spesific http method, and resource is before any base url or wildcards", done => {
```
Comment (jasonnutter) [inline lib/msal-angular/src/msal.interceptor.spec.ts:324]:
```suggestion
  it("does not attach authorization header when scopes set to null on specific http method, and resource is before any base url or wildcards", done => {
```

File: lib/msal-angular/src/msal.interceptor.ts [hunk 3]
```diff
@@ -207,12 +207,16 @@ export class MsalInterceptor implements HttpInterceptor {
...
-                            scopesForEndpoint.push(scope);
-                        });
+                        // Validate if scopes comes null to unprotect the resource in a certain http method 
+                        if (entry.scopes === null) {
+                            allMatchedScopes.push(null);
+                          } else {
+                            entry.scopes.forEach((scope) => {
+                              scopesForEndpoint.push(scope);
+                            });
+                          }
```
Comment (jasonnutter) [inline lib/msal-angular/src/msal.interceptor.ts:219]: Please dedent to match indendation of `if` block.

**Review by jasonnutter — CHANGES_REQUESTED** [review]
Thanks for teh PR, this looks like a good fix, left a few comments. Please cd to the root of the repo and run `npm run beachball:change`, fill out the prompts, and push the beachball change files. You can mark this as `minor`.

File: lib/msal-angular/angular.json (hunk 1 above)
Comment (at-DanielVargas) [reply lib/msal-angular/angular.json]: removed

File: lib/msal-angular/src/msal.interceptor.spec.ts (hunk 2 above)
Comment (at-DanielVargas) [reply lib/msal-angular/src/msal.interceptor.spec.ts]: Done, finger mistake

File: lib/msal-angular/src/msal.interceptor.ts (hunk 3 above)
Comment (at-DanielVargas) [reply lib/msal-angular/src/msal.interceptor.ts]: Done

File: change/@azure-msal-angular-9f9fd30b-1a85-4db7-bf2c-166af64eca00.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "allow unprotect resources on certainr http methods",
```
Comment (jasonnutter) [inline change/@azure-msal-angular-9f9fd30b-1a85-4db7-bf2c-166af64eca00.json:3]:
```suggestion
  "comment": "allow unprotect resources on certain http methods",
```

Comment (at-DanielVargas) [reply change/@azure-msal-angular-9f9fd30b-1a85-4db7-bf2c-166af64eca00.json]: done

File: change/@azure-msal-angular-9f9fd30b-1a85-4db7-bf2c-166af64eca00.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "allow unprotect resources on certain http methods",
```
Comment (jo-arroyo) [inline change/@azure-msal-angular-9f9fd30b-1a85-4db7-bf2c-166af64eca00.json:3]:
```suggestion
  "comment": "allow unprotect resources on certain http methods #3878",
```

**Review by jo-arroyo — APPROVED** [review]

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**